frontend/deb/debroot.go

```diff
@@ -452,6 +452,15 @@ func createInstallScripts(worker llb.State, spec *dalec.Spec, dir string) []llb.
 		}
 	}
 
+	if len(spec.Artifacts.InfoFiles) > 0 {
+		sorted := dalec.SortMapKeys(spec.Artifacts.InfoFiles)
+		for _, key := range sorted {
+			cfg := spec.Artifacts.InfoFiles[key]
+			resolved := cfg.ResolveName(key)
```

Comment:
I notice that in some cases (e.g. docs, licenses) we do this check:
```go
if resolved != key || cfg.SubPath != ""
```

Can you help me understand the reason for that check and why it's not needed here?